up-transport-vsomeip/src/transport_inner.rs

```diff
@@ -0,0 +1,59 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::storage::UPTransportVsomeipStorage;
+use async_trait::async_trait;
+use std::sync::Arc;
+use up_rust::{UListener, UMessage, UStatus, UUri};
+
+pub mod transport_inner_engine;
+pub mod transport_inner_handle;
+
+pub const UP_CLIENT_VSOMEIP_TAG: &str = "UPClientVsomeipInner";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_APP_EVENT_LOOP: &str = "app_event_loop";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL: &str = "register_listener_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL: &str = "send_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_INITIALIZE_NEW_APP_INTERNAL: &str =
+    "initialize_new_app_internal";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_START_APP: &str = "start_app";
+pub const UP_CLIENT_VSOMEIP_FN_TAG_STOP_APP: &str = "stop_app";
+
+const INTERNAL_FUNCTION_TIMEOUT: u64 = 3;
+/// Trait to make testing of [crate::UPTransportVsomeip] more decoupled and simpler.
+#[async_trait]
+pub(crate) trait MockableUPTransportVsomeipInner: Sync + Send {
```

Comment:
why is this called `Mockable...`?

Reply:
Fair... lemme revisit this. I think the name gives away why, but maybe not a great name.

Reply:
Based on that argument, almost every trait name should have that prefix ;-)

Reply:
Renamed this one, but I have a couple more Mockable... which I am having a hard time coming up with a name for.

Essentially wanted a way to be able to mock these registries that didn't dip into and have to use the proc macro generated extern "C" fns.

Going to love those ones as-is for now. :thinking: